opentelemetry-sdk/src/opentelemetry/sdk/_metrics/metric_reader_storage.py

```diff
@@ -12,23 +12,105 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Iterable
+from threading import RLock
+from typing import Dict, Iterable, List
 
+from opentelemetry._metrics.instrument import Instrument
 from opentelemetry.sdk._metrics.aggregation import AggregationTemporality
 from opentelemetry.sdk._metrics.measurement import Measurement
 from opentelemetry.sdk._metrics.point import Metric
 from opentelemetry.sdk._metrics.sdk_configuration import SdkConfiguration
+from opentelemetry.sdk._metrics.view import View
+
+
+# TODO: #2300
+class ViewStorage:
+    def __init__(
+        self,
+        view: View,
+        instrument: Instrument,
+        sdk_config: SdkConfiguration,
+    ) -> None:
+        pass
+
+    def consume_measurement(self, measurement: Measurement) -> None:
+        pass
+
+    def collect(self, temporality: AggregationTemporality) -> Iterable[Metric]:
+        pass
 
 
-# TODO: #2378
 class MetricReaderStorage:
     """The SDK's storage for a given reader"""
 
     def __init__(self, sdk_config: SdkConfiguration) -> None:
-        pass
+        self._lock = RLock()
+        self._sdk_config = sdk_config
+        self._view_storages: Dict[Instrument, List[ViewStorage]] = {}
+
+    def _get_or_init_view_storage(
```

Comment:
Single use method, please move this to where it is being used, it is hard to follow if we have to scroll up to find this code.

Reply:
i'm ok with this being a separate method as it makes it clearer what's happening in this code.

Reply:
Sure, resolving.